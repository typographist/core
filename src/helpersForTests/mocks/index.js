const breakpointsModel = require('./breakpointsModel');
const userConfig = require('./userConfig');

module.exports = {
  breakpointsModel,
  userConfig,
};
