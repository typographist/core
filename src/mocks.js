// @flow

type Breakpoint = {|
  base: number | number[],
  lineHeight: number,
  name: string,
  ratio: number,
  root: number,
  value: string,
|};

export type Breakpoints = Breakpoint[];

export const breakpoints: Breakpoints = [
  {
    base: [16, 33],
    lineHeight: 1.5,
    name: 'default',
    ratio: 1.1342044031840226,
    root: 12,
    value: '0px',
  },
  {
    base: 17,
    lineHeight: 1.5,
    name: 'tablet',
    ratio: 1.1258335640240535,
    root: 13,
    value: '640px',
  },
  {
    base: 18,
    lineHeight: 1.7,
    name: 'desktop',
    ratio: 1.333,
    root: 16,
    value: '1024px',
  },
  {
    base: 20,
    lineHeight: 1.7,
    name: 'lgDesktop',
    ratio: 1.333,
    root: 17,
    value: '1200px',
  },
  {
    base: 22,
    lineHeight: 1.7,
    name: 'xlDesktop',
    ratio: 1.333,
    root: 19,
    value: '1600px',
  },
];

export const userConfig = {
  base: ['1em', '2em'],
  lineHeight: 1.5,
  ratio: '45px at 6',
  tablet: {
    value: '40em',
    base: '17px',
  },
  desktop: {
    value: '64em',
    base: '18px',
    lineHeight: 1.7,
    ratio: 1.333,
  },
  lgDesktop: {
    value: '75em',
    base: '20px',
  },
  xlDesktop: {
    value: '100em',
    base: '22px',
  },
};
