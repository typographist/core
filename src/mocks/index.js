export { breakpoints, breakpointsModel } from './breakpoints';
export { userConfig, invalidUserConfig } from './user-configs';
