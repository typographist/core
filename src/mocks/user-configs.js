export const userConfig = {
  base: ['16px', '32px'],
  lineHeight: 1.5,
  ratio: '45px at 6',
  tablet: {
    breakpoint: '768px',
    base: '17px',
  },
  desktop: {
    breakpoint: '992px',
    base: '18px',
    lineHeight: 1.7,
    ratio: 1.333,
  },
  lgDesktop: {
    breakpoint: '1200px',
    base: '20px',
  },
  xlDesktop: {
    breakpoint: '1600px',
    base: '22px',
  },
};

export const invalidUserConfig = {
  base: ['1rem', '2em'],
  lineHeight: 1.5,
  ratio: '45 at 6',
  tablet: {
    base: '17px',
  },
  desktop: {
    breakpoint: '64rem',
    base: '18px',
    lineHeight: '1.7',
    ratio: 1.333,
  },
  lgDesktop: {
    breakpoint: '75rem',
    base: '20px',
  },
  xlDesktop: {
    breakpoint: '100rem',
  },
};
