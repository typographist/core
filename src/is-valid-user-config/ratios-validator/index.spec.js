import { UserConfig } from '../../mocks';
import { getRatios, isValidRatio, ratioIsVAlid, validateFields } from '.';

describe('user config validators', () => {
  describe('ratios validator', () => {
    describe('getRatios function', () => {
      it('returns all values of the ratio', () => {
        expect(getRatios(UserConfig)).toEqual(['45px at 6', 1.333]);
      });
    });

    describe('isValidRatio function', () => {
      it('if ratio contains the font size in pixels', () => {
        expect(isValidRatio('22px at 4')).toEqual(true);
      });

      it('if ratio contains the font size in ems', () => {
        expect(isValidRatio('22em at 4')).toEqual(true);
      });

      it('if ratio contains the `at` word', () => {
        expect(isValidRatio('22em at 4')).toEqual(true);
      });

      it('ratio contains the step', () => {
        expect(isValidRatio('1em at -4')).toEqual(true);
      });
    });

    describe('ratioIsVAlid function', () => {
      it('if ratio is the number', () => {
        expect(ratioIsVAlid(1.333)).toEqual(true);
      });

      it('if ratio is a valid string', () => {
        expect(ratioIsVAlid('33px at 5')).toEqual(true);
      });

      it('invalid ratio', () => {
        expect(ratioIsVAlid).toThrowErrorMatchingSnapshot();
      });
    });

    describe('validateFields function', () => {
      it('is all ratios are valid', () => {
        expect(validateFields(UserConfig)).toEqual(true);
      });
    });
  });
});
