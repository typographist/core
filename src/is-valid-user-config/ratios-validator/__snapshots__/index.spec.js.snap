// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user config validators ratios validator ratioIsVAlid function invalid ratio 1`] = `"Typographist >>> Check your user config. ratio:  must contain a number or a string containing the font size, the word 'at' and step. For example base: 1.25 or base: '36px at 6'."`;
